### PR TITLE
Add group issue boards to issue board view

### PR DESCRIPTION
**Pull request by tommyalatalo**
## Description
This adds the ability to choose group level boards when viewing issue
boards. Some additional flags such as label filtering are also added
since the Gitlab issue boards are mainly just the column layouts
including all issues in the project/group. "Opened" and "Closed" issue lists are now included in the view.

## Related Issue
https://github.com/profclems/glab/issues/840

## How Has This Been Tested?
Manually tested using existing group and project boards on a private Gitlab instance.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation
- [ ] Chore (Related to CI or Packaging to platforms)

### CODE REVIEW

File: commands/issue/board/view/issue_board_view.go
Comment (zemzale) [inline commands/issue/board/view/issue_board_view.go]: The way how these variables are declared and used in the command is considered "legacy" in glab. 

If you look at other commands like `mr list` we define an options struct that holds all the options and some other misc stuff.

Could you maybe convert this command to use this kind of options struct? It has made everything cleaner for us.

https://github.com/profclems/glab/blob/trunk/commands/mr/list/mr_list.go#L21-L52

File: commands/issue/board/view/issue_board_view.go
Comment (zemzale) [inline commands/issue/board/view/issue_board_view.go]: Can we make this as an `else` for the previous `if` 
```go
 	if boardInfo[selectedBoard].group != nil {
 ```
 
It was done like that some lines above(line 184) so we should just keep to that convention.

**Review by zemzale — COMMENTED** [review]
Nice work @tommyalatalo 

I left a few nitpiky comments. Before merging I would want to have a review from @profclems since this is quite a big MR.

File: commands/issue/board/view/issue_board_view.go
Comment (tommyalatalo) [reply commands/issue/board/view/issue_board_view.go]: Pushed this now, see if it does the job, I agree that it's quite nice to have them in a struct like that, I'll remember that for future projects.

File: commands/issue/board/view/issue_board_view.go
Comment (tommyalatalo) [reply commands/issue/board/view/issue_board_view.go]: I did away with all the 'else' statements in the file, I don't know why my mind slipped to add it there as I make an effort to avoid them for clarity.

File: commands/issue/board/view/issue_board_view.go
Comment (profclems) [inline commands/issue/board/view/issue_board_view.go]: I'm not super happy about these flags. It would be best to have one single flag to accept the assignee username

Like the flag in `mr creae`: https://github.com/profclems/glab/blob/9b44934c07d1a17d19fd26d5886a752d7cf06a30/commands/mr/create/mr_create.go#L162

You can use the `api.UsersByNames` function as used here:
https://github.com/profclems/glab/blob/9b44934c07d1a17d19fd26d5886a752d7cf06a30/commands/mr/create/mr_create.go#L505-L512

**Review by profclems — CHANGES_REQUESTED** [review]
Thanks a lot for working on this @tommyalatalo!

All looks good except for one issue we have to fix before we merge this.

**Comment by tommyalatalo** [comment]
@profclems I finally came back to this after a long time.
I've update the PR with some fixes, a pretty big refactoring to split the unwieldy command into smaller chunks, and made the board view styling more readable.

Have a look at this again when you have the chance and hopefully we can merge and close this sooner rather than later.

**Comment by tommyalatalo** [comment]
What is the status on this? Haven't heard anything on this after pushing an updated solution.

**Comment by profclems** [comment]
Hi @altosys, thanks a lot for this PR. I love this, however, we announced that this project is officially adopted by GitLab https://github.com/profclems/glab/issues/983#issue-1175723125. All new contributions should be made to the GitLab repo: https://gitlab.com/gitlab-org/cli

**Pull request closed without merging**
